staging/src/k8s.io/code-generator/cmd/validation-gen/validators/each.go

```diff
@@ -275,33 +281,58 @@ func (evtv eachValTagValidator) getListValidations(fldPath *field.Path, t *types
 	result := Validations{}
 	result.OpaqueValType = validations.OpaqueType
 
-	var listMap *listMap
+	var listMetadata *listMetadata
 	if lm, found := evtv.byFieldPath[fldPath.String()]; found {
-		if !lm.declaredAsMap {
-			return Validations{}, fmt.Errorf("found listMapKey without listType=map")
+		if lm.declaredAsSet && lm.declaredAsMap {
```

Comment:
This exposes a bug.  If you have a field which sets `listType=map` and never sets a key, you don't get an erro UNLESS you have an `eachVal`.  Without that, it's not considered an error because this code never runs.

OK for now, but I will think how to fix this.  It may require a different kind of validator.